clientv3/client.go

```diff
@@ -157,16 +159,25 @@ func (c *Client) Dial(endpoint string) (*grpc.ClientConn, error) {
 		grpc.WithBlock(),
 		grpc.WithTimeout(c.cfg.DialTimeout),
 	}
-	if c.creds != nil {
-		opts = append(opts, grpc.WithTransportCredentials(*c.creds))
-	} else {
-		opts = append(opts, grpc.WithInsecure())
-	}
 
 	proto := "tcp"
-	if url, uerr := url.Parse(endpoint); uerr == nil && url.Scheme == "unix" {
-		proto = "unix"
-		// strip unix:// prefix so certs work
+	creds := c.creds
+	if url, uerr := url.Parse(endpoint); uerr == nil && strings.Contains(endpoint, "://") {
+		switch url.Scheme {
+		case "unix":
+			proto = "unix"
+		case "http":
+			creds = nil
+		case "https":
+			if creds == nil {
+				tlsconfig := &tls.Config{InsecureSkipVerify: true}
```

Comment:
Thought about it some more-- if there's a TLS config in the client config then it'll use that as the credentials instead of creating an empty TLS for the https:// scheme case. I don't think verifying the certs makes sense if there's no given TLS config since there's no user-provided CA to compare against anyway.